src/core/Header/HeaderButton.tsx

```diff
@@ -78,11 +87,21 @@ export const HeaderButton = (props: HeaderButtonProps) => {
     ...rest,
   };
 
-  return menuItems ? (
-    <DropdownButton {...buttonProps} menuItems={menuItems} />
-  ) : (
-    <Button {...buttonProps} />
-  );
+  if (menuItems && props.onClick) {
+    return (
+      <SplitButton
+        {...buttonProps}
+        menuItems={menuItems}
+        onClick={() => buttonProps.onClick}
```

Comment:
Just write:
```tsx
onClick={buttonProps.onClick}
```

Reply:
Do I need to add something else? I'm getting this error:
![image](https://user-images.githubusercontent.com/83585998/129355956-1aa6cbb9-4c0b-4fa2-9f27-f28fdb48f5aa.png)

Reply:
You can set the correct type for SplitButton onClick 😃 It should be `onClick: MouseEventHandler<HTMLButtonElement>`

Reply:
Once this type is fixed, I think we'll be able to use it just from `...buttonProps`.

Reply:
We tried to pass only `{...buttonProps}` but it seems that Typescript doesn't understand that `onClick` and `menuItems` exist inside `buttonProps`, it only is happy when props are passed explicitly.
If you know the way, we are happy to see it 😃

Reply:
Type guard will help here
![image](https://user-images.githubusercontent.com/9084735/129572373-8a1e87b3-7c2e-4e12-98f2-d6738f594730.png)